src/actions/users.js

```diff
@@ -208,15 +214,15 @@ function completeLogin(data) {
             }
         }
         if (roles.size > 0) {
-            dispatch(loadRolesIfNeeded(roles));
+            dispatch(loadRolesIfNeeded([...roles]));
```

Comment:
I don't understand this change: this forces a duplication of `roles`. What makes this necessary?

Reply:
Yep. this part is necessary for:

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ src/actions/users.js:217:40

Cannot call loadRolesIfNeeded with roles bound to roles because Set [1] is incompatible with array type [2].

     src/actions/users.js
 [1] 207│         const roles = new Set();
        :
     214│             }
     215│         }
     216│         if (roles.size > 0) {
     217│             dispatch(loadRolesIfNeeded(roles));
     218│         }
     219│
     220│         return {data: true};

     src/actions/roles.js
 [2]  62│ export function loadRolesIfNeeded(roles: Array<string>): ActionFunc {
```

Reply:
Aha, I missed the nuance that `roles` was a `Set`. I suppose flow doesn't model a supertype that both an `Array` and `Set` could match against for use on `loadRolesIfNeeded`?

Reply:
I like it! Your call on whether to do it now or in the future :)